test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx

```diff
@@ -1,195 +1,218 @@
+    }
+
+
+    async componentDidUpdate( prevProps ) {
+        if ( prevProps.buildSelected !== this.props.buildSelected ) {
+            await this.updateData();
+        }
+    }
+
+    async updateData() {
+
+        const { buildSelected } = this.props;
+        const buildName = encodeURIComponent( buildSelected );
+        const response = await fetch( `/api/getBuildHistory?type=Perf&buildName=${buildName}&status=Done&limit=100&asc`, {
+            method: 'get'
+        } );
+        const results = await response.json();
+        const resultsByJDKBuild = [];
+        let baseLine = [];
+        
+        results.forEach(( t, i ) => {
+            if (t.buildResult == "SUCCESS" || t.validAggregateInfo) {
+                for ( let element of t.aggregateInfo ) {
+                    const jdkDate = t.timestamp;
+                    if(!resultsByJDKBuild[t.buildName]) {
+                        resultsByJDKBuild[t.buildName] = {} ;
+                    }
+                    for( let singleMetric of element.metrics) {
+                        if(!resultsByJDKBuild[t.buildName][singleMetric.name]) {
+                            resultsByJDKBuild[t.buildName][singleMetric.name] = {};
+                        }
+                        resultsByJDKBuild[t.buildName][singleMetric.name][jdkDate] = resultsByJDKBuild[t.buildName][singleMetric.name][jdkDate] || [];
+                        resultsByJDKBuild[t.buildName][singleMetric.name][jdkDate].push( {
+                            [singleMetric.name]: singleMetric.statValues["mean"],
+                            additionalData: {
+                                mean: singleMetric.statValues["mean"],
+                                max: singleMetric.statValues["max"],
+                                min: singleMetric.statValues["min"],
+                                median: singleMetric.statValues["median"],
+                                stddev: singleMetric.statValues["stddev"],
+                                CI: singleMetric.statValues["CI"],
+                                validIterations: singleMetric.statValues["validIterations"],
+                                testId: (Array.isArray(t.tests) && t.tests.length > 0) ? t.tests[0]._id : null,
+                                hasChildren: t.hasChildren,
+                                parentId:t._id,
+                                buildName: t.buildName,
+                                buildNum: t.buildNum,
+                                javaVersion: t.javaVersion,
+                            },
+                        } );
+                    }
+                }
+            }
+        } );
+        let baseLineData = [];
+        let metric_data = [];
+        for ( let singleTestName in resultsByJDKBuild) {
+            if (!metric_data[singleTestName]){
+                metric_data[singleTestName] = [];
+            }
+            for ( let singleMetricName in resultsByJDKBuild[singleTestName]){
```

Comment:
Same thing as before. Maybe just call it `metric` instead of `singleMetricName`.